js/pro/deribit.js

```diff
@@ -0,0 +1,843 @@
+            };
+            const handler = this.safeValue (handlers, channelId);
+            if (handler !== undefined) {
+                return handler.call (this, client, message);
+            }
+            throw new NotSupported (this.id + ' no handler found for this message ' + this.json (message));
+        }
+        const result = this.safeValue (message, 'result', {});
+        const accessToken = this.safeString (result, 'access_token');
+        if (accessToken !== undefined) {
+            return this.handleAuthenticationMessage (client, message);
+        }
+        return message;
+    }
+
+    handleAuthenticationMessage (client, message) {
+        //
+        //     {
+        //         jsonrpc: '2.0',
+        //         id: 1,
+        //         result: {
+        //             token_type: 'bearer',
+        //             scope: 'account:read_write block_trade:read_write connection custody:read_write mainaccount name:ccxt trade:read_write wallet:read_write',
+        //             refresh_token: '1686927372328.1EzFBRmt.logRQWXkPA1oE_Tk0gRsls9Hau7YN6a321XUBnxvR4x6cryhbkKcniUJU-czA8_zKXrqQGpQmfoDwhLIjIsWCvRuu6otbg-LKWlrtTX1GQqLcPaTTHAdZGTMV-HM8HiS03QBd9MIXWRfF53sKj2hdR9nZPZ6MH1XrkpAZPB_peuEEB9wlcc3elzWEZFtCmiy1fnQ8TPHwAJMt3nuUmEcMLt_-F554qrsg_-I66D9xMiifJj4dBemdPfV_PkGPRIwIoKlxDjyv2-xfCw-4eKyo6Hu1m2h6gT1DPOTxSXcBgfBQjpi-_uY3iAIj7U6xjC46PHthEdquhEuCTZl7UfCRZSAWwZA',
+        //             expires_in: 31536000,
+        //             access_token: '1686923272328.1CkwEx-u.qHradpIulmuoeboKMEi8PkQ1_4DF8yFE2zywBTtkD32sruVC53b1HwL5OWRuh2nYAndXff4xuXIMRkkEfMAFCeq24prihxxinoS8DDVkKBxedGx4CUPJFeXjmh7wuRGqQOLg1plXOpbF3fwF2KPEkAuETwcpcVY6K9HUVjutNRfxFe2TR7CvuS9x8TATvoPeu7H1ezYl-LkKSaRifdTXuwituXgp4oDbPRyQLniEBWuYF9rY7qbABxuOJlXI1VZ63u7Bh0mGWei-KeVeqHGNpy6OgrFRPXPxa9_U7vaxCyHW3zZ9959TQ1QUMLWtUX-NLBEv3BT5eCieW9HORYIOKfsgkpd3'
+        //         },
+        //         usIn: '1655391872327712',
+        //         usOut: '1655391872328515',
+        //         usDiff: 803,
+        //         testnet: false
+        //     }
+        //
+        const future = this.safeValue (client.futures, 'authenticated');
+        if (future !== undefined) {
+            future.resolve (true);
+        }
+        return message;
+    }
+
+    async authenticate (params = {}) {
+        const url = this.urls['api']['ws'];
+        const client = this.client (url);
+        const time = this.milliseconds ();
+        const timeString = this.numberToString (time);
+        const nonce = timeString;
+        const messageHash = 'authenticated';
+        const future = client.future ('authenticated');
+        const authenticated = this.safeValue (client.subscriptions, messageHash);
+        if (authenticated === undefined) {
+            this.checkRequiredCredentials ();
+            const requestId = this.requestId ();
+            const signature = this.hmac (timeString + '\n' + nonce + '\n', this.secret, 'sha256');
```

Comment:
we're missing `encode` here otherwise breaks the Python version
![image](https://user-images.githubusercontent.com/43336371/195866398-d4491dc1-54f9-4d92-b6e6-e867cdebad13.png)

Reply:
great catch!